tests/matrix.rs

```diff
@@ -365,3 +365,29 @@ fn test_from_angle() {
     let rot3: Matrix2<f64> = Matrix2::from_angle(rad(f64::consts::PI));
     assert!((rot3 * Vector2::new(1.0, 1.0)).approx_eq(&Vector2::new(-1.0, -1.0)));
 }
+
+#[test]
+fn test_matrix3_into_quaternion()
+{
+    fn eq(a: Matrix3<f32>, b: Matrix3<f32>) {
+        if !a.approx_eq(&b) {
+            panic!("{:?} != {:?}", a, b);
+        }
+    }
+
+    fn test_angles(x: Rad<f32>, y: Rad<f32>, z: Rad<f32>) {
+        let basis: Basis3<f32> = Rotation3::from_euler(x, y, z);
+        let quaternion: Quaternion<f32> = basis.into();
+        eq(basis.into(), quaternion.into());
```

Comment:
You can use the [`approx_eq!`](https://github.com/bjz/cgmath-rs/blob/master/src/approx.rs#L64) macro for this.